crates/wysiwyg/src/tests/test_suggestions.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{MenuAction, TrailingStrategy};
+
+use super::testutils_composer_model::{cm, tx};
+
+#[test]
+fn test_replace_text_suggestion() {
+    let mut model = cm("|");
+    let update = model.replace_text("/".into());
+    let MenuAction::Suggestion(suggestion) = update.menu_action else {
+        panic!("No suggestion pattern found")
+    };
+    model.replace_text_suggestion("/invite".into(), suggestion);
+    assert_eq!(tx(&model), "/invite&nbsp;|");
+}
+
+#[test]
+fn test_set_link_suggestion() {
+    let mut model = cm("|");
+    let update = model.replace_text("@alic".into());
+    let MenuAction::Suggestion(suggestion) = update.menu_action else {
+        panic!("No suggestion pattern found")
+    };
+    model.set_link_suggestion(
+        "https://matrix.to/#/@alice:matrix.org".into(),
+        "Alice".into(),
+        suggestion,
+    );
+    assert_eq!(
+        tx(&model),
+        "<a href=\"https://matrix.to/#/@alice:matrix.org\">Alice</a>&nbsp;|",
+    );
+}
+
+#[test]
+fn test_set_link_suggestion_with_colon_space_trailing() {
+    let mut model = cm("|");
+    let update = model.replace_text("@alic".into());
+    let MenuAction::Suggestion(mut suggestion) = update.menu_action else {
+        panic!("No suggestion pattern found")
+    };
+    suggestion.trailing_strategy = TrailingStrategy::ColonSpace;
+    model.set_link_suggestion(
+        "https://matrix.to/#/@alice:matrix.org".into(),
+        "Alice".into(),
+        suggestion,
+    );
+    assert_eq!(
+        tx(&model),
+        "<a href=\"https://matrix.to/#/@alice:matrix.org\">Alice</a>:&nbsp;|",
```

Comment:
One question for design: is the trailing colon something we're committed to for the new composer?

I always found it a bit awkward when I wanted to mention multiple users before writing my message. For example:

```
@user1, @user2 what are your thoughts?
```

Reply:
Yeah that's a good question, I'll leave it at that for now, and we can discuss it with the team.

Reply:
Actually done here: https://github.com/matrix-org/matrix-rich-text-editor/pull/591/commits/f8afc7d5b04e282fa7d6f0f637fb40664d6f1229